src/adapters/refract/transformResource.coffee

```diff
@@ -0,0 +1,122 @@
+_ = require('./helper')
+blueprintApi = require('../../blueprint-api')
+getDescription = require('./getDescription')
+getHeaders = require('./getHeaders')
+getUriParameters = require('./getUriParameters')
+
+module.exports = (resourceElement) ->
+  resources = []
+
+  resourceDescription = getDescription(resourceElement)
+
+  _.transitions(resourceElement).forEach((transitionElement) ->
+    description = getDescription(transitionElement)
+
+    resourceParameters = getUriParameters(_.get(resourceElement, 'attributes.hrefVariables'))
+    actionParameters = getUriParameters(_.get(transitionElement, 'attributes.hrefVariables'))
+
+    attributes = _.dataStructures(resourceElement)
+    attributes = undefined if _.isEmpty(attributes)
```

Comment:
Sorry, didn't see your comments. They didn't pop up until I refreshed (this always bugs me!).